src/native/libs/System.Native/pal_threading.c

```diff
@@ -233,6 +233,14 @@ int32_t SystemNative_CreateThread(uintptr_t stackSize, void *(*startAddress)(voi
     error = pthread_attr_setdetachstate(&attrs, PTHREAD_CREATE_DETACHED);
     assert(error == 0);
 
+#if defined(HOST_OSX) || defined(HOST_MACCATALYST)
```

Comment:
I think we can do it all Apple platforms, including mobile ones, given that Mono has larger stack size on those platforms too.